src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php

```diff
@@ -132,6 +132,14 @@ public function reverseTransform($value)
             throw new TransformationFailedException('Expected a string.');
         }
 
+        // handle seconds ignored by user's browser when seconds as single_text is 0
+        if ($this->parseFormat === 'H:i:s|') {
+            if (!preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9])(?::[0-5][0-9]))', $value) &&
+                preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9]))', $value)) {
```

Comment:
Yes, fixing it pre_submit makes more sense. I think it's considerable to add this to built-in `TimeType`.

However i still dont understand the 2 regexes.. you could simply use

```php
if (preg_match('/^(?:[0-1][0-9]|[2][0-3]|[0-9]):[0-5][0-9]$/', $time)) {
  $time .= ':00';
}
```
right?